host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template

```diff
@@ -10,7 +10,8 @@ ENV AzureWebJobsScriptRoot=/home/site/wwwroot \
     HOME=/home \
     FUNCTIONS_WORKER_RUNTIME=dotnet \
     DOTNET_USE_POLLING_FILE_WATCHER=true \
-    HOST_VERSION=${HOST_VERSION}
+    HOST_VERSION=${HOST_VERSION} \
+    ASPNETCORE_CONTENTROOT=/azure-functions-host
```

Comment:
From https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-6.0#content-root, this should default to the folder where the app assembly resides.
Since we are not setting this currently , in these images shouldnt that already be /azure-functions-host ?

Reply:
very likely this is a bug - needs follow up with .net team - file watchers count drastically went down after setting this explicitly. @balag0  very like need to set this in mesh images as well until we figure this out in how to set this in functions host code.

Reply:
Just a quick guess (but would be good to validate), based on https://github.com/aspnet/MetaPackages/blob/master/src/Microsoft.AspNetCore/WebHost.cs#L152
if the env. variable is not set, it defaults to current directory.
In the dockerfiles there is no WORKDIR is set. so the current directory at that point could be /

If we have a way to repro/validate, can we add 
WORKDIR /azure-functions-host
to the dockerfiles and not set ASPNETCORE_CONTENTROOT explicitly
and see if that fixes the issue?

Reply:
good idea. @CooperLink - it is worth it to setup a repro for this on private stamp - host image without this change and Storage app setting - mount Azure Files with large folder structure.

Reply:
After pushing hotfix changes to UD6 I will be setting up a private stamp repro for this to test the above. Thank you both for your comments

Reply:
@balag0  - @CooperLink  brought up a good point - would setting WorkingDir affect applications? I remember we considered this inthe past and decided was a breaking change - but OK to do this in V4 while it was in preview.

Reply:
Yes, had a chat with Cooper just now as well. Yes agree that using the appsetting is the safer and better fix. Setting the WORKDIR was just an experiment to make sure we understand the behavior (in case there was a bug in the framework) :)

Also here is an approach to set this in the code (but dockerfiles is still better fix for I think)
https://stackoverflow.com/questions/59033413/c-sharp-aspnetcore-set-content-root-path

Reply:
@balag0 - is this change needed in mesh images?

Reply:
Potentially since the setup is the same, but given we dont have any CRIs and the underlying issue here was exposed only after a config update on workers, I think its not urgent and we can wait until these images are rolled out and tested in case we identify any other scenarios to consider.